src/nvim/eval.c

```diff
@@ -11686,14 +11691,70 @@ static void f_jobstart(typval_T *argvars, typval_T *rettv, FunPtr fptr)
         return;
       }
     }
+    dictitem_T *item;
+    // or new_env
+    item = tv_dict_find(job_opts, S_LEN("env"));
+    if (item) {
+      size_t env_size = 0; // len ?
+      size_t i = 0;
+      // size_t orig_env_size = 0;
+      if (item->di_tv.v_type != VAR_DICT) {
+        EMSG2(_(e_invarg2), "Expected a dictionary for 'env'");
+        return;
+      }
+
+      // number of items in dict
+      env_size += (size_t)tv_dict_len(item->di_tv.vval.v_dict);
+      ILOG("%d passed env variables", env_size);
+
+      if (!reset_env) {
+        char **environ = os_getfullenv();
+        size_t current_env_size = 0; // = strlen(environ); // strlen excludes null byte
+        for (env = environ; *env; env++) {
+          current_env_size++;
+        }
+        ILOG("current env_len=%d", current_env_size);
+        env_size += current_env_size;
+        env = xmalloc( (env_size) * sizeof(char));
```

Comment:
`sizeof(char)` is guaranteed to be 1 by C99 standard itself. Also don’t use `sizeof(type)` as a general rule, use variables: in your case it would be `sizeof(*env)`, which is also size of the *pointer* `char *` like it should be, with `sizeof(char)` it should SEGV.